### PR TITLE
Rework env

### PR DESCRIPTION
**Pull request by felixgabler**
We can simplify the logic around Env by quite a lot I believe 
I began working on it a bit but could use some comments about best practices etc.

### CODE REVIEW

**Comment by felixgabler** [comment]
I ran scalafmt before most commits.

**Comment by zy4** [comment]
Also the build fails...

**Comment by zy4** [comment]
why are there so many changes in the imports? especially formatting and deletions?

**Comment by felixgabler** [comment]
It fails with the following entry in the log:
value asJson is not a member of io.circe.JsonObject

From that I figure that the build process is not fully working on our Travis instance.

**Comment by felixgabler** [comment]
The import changes happened after scalafmt

**Comment by zy4** [comment]
and the scalajs related deletions?

**Comment by zy4** [comment]
Normally these files shouldn't be touched at all. I think something must be broken in this branch. Gonna check it later.

**Comment by zy4** [comment]
you need to rebase first. you are merging from a branch where scalajs is partially deleted already.

File: build.sbt [hunk 1]
```diff
@@ -186,6 +186,7 @@ fork := true // required for "sbt run" to pick up javaOptions
 javaOptions += "-Dplay.editor=http://localhost:63342/api/file/?file=%s&line=%s"
 fork in Test := true
 logLevel in Test := Level.Info
+scalafmtOnCompile := true
```
Comment (zy4) [inline build.sbt]: please remove, this is slowing down everything

**Comment by zy4** [comment]
So far it looks reasonable to me but it should first be tested by running a few jobs which i didn't.

**Comment by vikramalva** [comment]
/cc @felixgabler job submission does not work. HOSTNAME is set correctly in application.conf

---
2019-03-01 10:15:52,607 [warn] r.c.n.ChannelFactory - [Supervisor-1/Connection-2] Skip channel init as host is null
2019-03-01 10:16:21,775 [error] a.d.Dispatcher - No configuration setting found for key 'HOSTNAME'
com.typesafe.config.ConfigException$Missing: No configuration setting found for key 'HOSTNAME'
	at com.typesafe.config.impl.SimpleConfig.findKeyOrNull(SimpleConfig.java:156)
	at com.typesafe.config.impl.SimpleConfig.findOrNull(SimpleConfig.java:174)
	at com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:188)
---

**Comment by felixgabler** [comment]
@vikramalva Please test it again.

**Comment by felixgabler** [comment]
Something weird was going on with travis. I pushed a renamed version of rework_env and now finally the build went through. Now we can merge after review.

**Pull request merged**